Antidote/AbstractCallViewController.h

```diff
@@ -0,0 +1,34 @@
+//
+//  AbstractCallViewController.h
+//  Antidote
+//
+//  Created by Chuong Vu on 7/2/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@class OCTSubmanagerCalls;
+@class OCTChat;
+@class OCTCall;
+
+@interface AbstractCallViewController : UIViewController
+
+/**
+ * Create an instance of the CallViewController.
+ * @param chat Appropriate chat for the call.
+ * @param manager The call maanger for the application.
+ */
+- (instancetype)initWithChat:(OCTChat *)chat submanagerCalls:(OCTSubmanagerCalls *)manager;
+
+/**
+ * The call associated with this view.
+ */
+@property (strong, nonatomic) OCTCall *call;
+
+/**
+ * The call manager responsible for call handling.
+ */
+@property (weak, nonatomic) OCTSubmanagerCalls *manager;
```

Comment:
No need for readwrite properties in public interface. You can add `readonly` here and leave `readwrite` in .m file